src/Entities/Player.cpp

```diff
@@ -1046,12 +1046,21 @@ void cPlayer::Respawn(void)
 	m_LifetimeTotalXp = 0;
 	// ToDo: send score to client? How?
 
-	m_ClientHandle->SendRespawn(GetWorld()->GetDimension(), true);
+	m_ClientHandle->SendRespawn(dimOverworld, true);
 
 	// Extinguish the fire:
 	StopBurning();
 
-	TeleportToCoords(GetLastBedPos().x, GetLastBedPos().y, GetLastBedPos().z);
+	cWorld *world = GetWorld();
+	if (world->GetDimension() != dimOverworld)
+	{
+		world = cRoot::Get()->CreateAndInitializeWorld(world->GetLinkedOverworldName(), dimOverworld, world->GetLinkedOverworldName(), false);
```

Comment:
@NiLSPACE I'm against this bail out idea, because it's inconsistent with the nether portal code and it would only cause trouble, when the server has not loaded the overworld(endless lava deaths...).
Instead a new issue referencing the core <code>CreateAndInitializeWorld</code> problem should be created to fix all calls to this function.